client/scanner.js

```diff
@@ -211,6 +241,30 @@ class Scanner {
     }
   }
 
+  /**
+   * Watch for new transactions as they are created.
+   * @param {Number} fromBlock The block from which to begin watch.
+   * @returns {void}
+   */
+  async watchBlocks(fromBlock) {
+    const requestFactory = await this.eac.requestFactory();
+    this.requestWatcher = await requestFactory.watchRequests(fromBlock,
+      async (request) => {
+        if (!this.isCorrect(request)) return;
+
+        this.log.debug(`[${request}] Discovered.`)
+        if (!this.cache.has(request)) {
+          // If it's not already in cache, find windowStart.
+          const txRequest = await this.fill(request)
+
+          if (txRequest && await this.isExecutable(txRequest) ) {
+            // If the isExecutable returns True, store it.
+            this.store(txRequest)
```

Comment:
I think instead of storing it and waiting for the next `scanCache` interval, if we pick up an executable transaction request here we should just route it to be executed right away.

Reply:
Done:
https://github.com/adibas03/eac.js-client/blob/a418a8e772cc5af2c601ff92ef21f2a215cb4666/client/scanner.js#L252